### PR TITLE
add a filter feature

### LINKED ISSUE #18: Filter by Alphabetical Order
**Is your feature request related to a problem? Please describe.**
Allow users to search Pokemon in alphabetical order.

### PR DESCRIPTION
**Pull request by JayDampitan**
## Changes
1. Added filtering feature for Pokémon's

## Purpose
Allows the users to filter Pokémon by alphabetical order

## Approach
Wrote two functions to handle the sorting. Used the radio type button to handle the filtering. 

Closes #18

### CODE REVIEW

**Review by connielion — APPROVED** [review]
Looks good.

Is the filter working? For Brain.JS, do you still need that button and console log to test filtering?

File: pokeapp/src/components/FilterBar.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        return 0;
+      })
+    );
+  };
+  const sortZtoA = () => {
+    setAllPokemon(
+      allPokemon.sort((pokemon1, pokemon2) => {
+        if (pokemon1.name < pokemon2.name) return 1;
+        if (pokemon1.name > pokemon2.name) return -1;
+        return 0;
```
Comment (kmausolf) [inline pokeapp/src/components/FilterBar.js:17]: I'd advise against overwriting the original copy.  The reason for this is that if you want to apply filters or sorting to the data, it's easier to remove them if you have a separate copy of the mutated data.  I recommend keeping the original copy static, then using that original copy to _derive_ the mutated version based on sort and filter.

Comment (kmausolf) [reply pokeapp/src/components/FilterBar.js:17]: In fact, if you apply filters then overwrite the original, you'll run into the problem that I fixed on a client project last year.  The team was applying filters to the original copy of the data, which made it impossible to remove the filters without performing a whole API call to restore the original.

**Review by BlackCubes — APPROVED** [review]
LGTM! But I would listen to the Keyslayer on his comment. A bug might happen. Maybe in some future issue? But good job!

**Pull request merged**